Source/Core/Common/TraversalServer.cpp

```diff
@@ -0,0 +1,458 @@
+// This file is public domain, in case it's useful to anyone. -comex
+
+// The central server implementation.
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include "TraversalProto.h"
+
+#define DEBUG 1
+
+static u64 currentTime;
+
+struct OutgoingPacketInfo
+{
+	TraversalPacket packet;
+	TraversalRequestId misc;
+	struct sockaddr_in6 dest;
+	int tries;
+	u64 sendTime;
+};
+
+template <typename T>
+struct EvictEntry
+{
+	u64 updateTime;
+	T value;
+};
+
+template <typename V>
+struct EvictFindResult
+{
+	bool found;
+	V* value;
+};
+
+template <typename K, typename V>
+EvictFindResult<V> EvictFind(typename std::unordered_map<K, EvictEntry<V>>& map, const K& key, bool refresh = false)
```

Comment:
`typename std::unordered_map<K, EvictEntry<V>>&` should be able to be shortened to `std::unordered_map<K, EvictEntry<V>>&`